content/howto/extensibility/how-to-create-a-native-pluggable-widget.md

```diff
@@ -0,0 +1,1344 @@
+
+   - App name: **GroupBoxTestProject**
+   - Enable online services: **No**
+
+   After that, click the **Create app** button.
+
+   Optionally you may remove all unused custom widgets to optimize the debugging process. Select **Project > Show Project Directory in Explorer** from the Mendix Studio Pro menu and open the **widgets** folder. Then, delete all the files in this folder.
+
+   After that, close the project by selecting **File > Close Project** from the top menu bar.
+
+2. Move the contents of the Mendix test project folder into **test/MxTestProject** which is located inside your group box widget project folder.
+
+3. Open the moved Mendix test project by selecting **File > Open Project...** from the top menu bar. In the **Open App** dialog box, select **Locally on disk** and browse to the **GroupBoxTestProject.mpr** file inside your group box widget project folder.
+
+4. Create a test case for the group box widget:
+
+   1. Open **Home_Native** page.
+   2. Copy "Home" container and paste it underneath that container
+   3. Delete the content.
+   4. Search in toolbox for the **Group Box** widget.
+   5. Drag the widget into the new container.
+   6. Double click the widget and fill in your name as default value.
+
+5. Setting up the test project is done. Run the test project locally and use the makeitnative app to view the sample widget code in action. If you followed the steps carefully you should see yellow text in your app saying "Hello {your name}".
+
+### 3.3 Building the group box widget
+
+Now that we have seen our project code running in native, it is time to transform our sample widget into an actual group box widget.
+
+Open up your group box widget project folder in your IDE of choice. From now on, all file references will be relative to this project folder.
+
+#### 3.3.1 Creating static UI
+
+First we will change the display component. A display component represents the widget and doesn't interact with Mendix specific APIs and, therefore, can be used in any React application. The current display component is located in **src/components** and has the filename **HelloWorldSample.tsx**. Open this file and you can see a typical React Native component with a render method. This method returns the Graphical User Inteface (GUI) of the component.
+
+Let's change the output of this method to display a group box. Replace the existing code with the following:
+
+```tsx
+return (
+  <View style={this.styles.container}>
+    <View style={this.styles.header}>
+      <Text style={this.styles.headerContent}>Header caption</Text>
+      <Text style={this.styles.headerContent}>-</Text>
+    </View>
+    <View style={this.styles.content}>Content</View>
+  </View>
+);
+```
+
+To apply a custom style to our structure above, replace the following code representing the default style:
+
+```tsx
+const defaultStyle: CustomStyle = {
```

Comment:
New chapter on style, explain rudimentary basics difference from web styling, link to internal and external docs
- not cascading
- flex based